frontend/src/components/common/Resource/DocsViewer.tsx

```diff
@@ -22,32 +21,54 @@ const useStyles = makeStyles(() => ({
 function DocsViewer(props: { docSpecs: any }) {
   const { docSpecs } = props;
   const classes = useStyles();
-  const [docs, setDocs] = React.useState<object | null>(null);
-  const [docsError, setDocsError] = React.useState<string | null>(null);
+  const [docs, setDocs] = React.useState<
+    (
+      | {
+          data: null;
+          error: any;
+          kind: string;
+        }
+      | {
+          data: any;
+          error: null;
+          kind: string;
+        }
+      | undefined
+    )[]
+  >([]);
+  const [docsLoading, setDocsLoading] = React.useState(false);
   const { t } = useTranslation('resource');
 
   React.useEffect(() => {
-    setDocsError(null);
-
-    if (docSpecs.error) {
-      t('Cannot load documentation: {{ docsError }}', { docsError: docSpecs.error });
-      return;
-    }
-    if (!docSpecs.apiVersion || !docSpecs.kind) {
-      setDocsError(
-        t(
-          'Cannot load documentation: Please make sure the YAML is valid and has the kind and apiVersion set.'
-        )
-      );
-      return;
-    }
-
-    getDocDefinitions(docSpecs.apiVersion, docSpecs.kind)
-      .then(result => {
-        setDocs(result?.properties || {});
+    setDocsLoading(true);
+    // fetch docSpecs for all the resources specified
+    Promise.allSettled(
+      docSpecs.map((docSpec: { apiVersion: string; kind: string }) => {
+        return getDocDefinitions(docSpec.apiVersion, docSpec.kind);
+      })
+    )
+      .then(values => {
+        //@ts-ignore
```

Comment:
I'd prefer to avoid ts-ignores. If you need help fixing the types, we can look into it together.